### DISCUSSION TITLE
Filtering items of listings based on meta data

### DISCUSSION THREAD

**Discussion by mllg** [quarto-dev/quarto-cli#2828] [Q&A]
I have a folder `publications` which contains a `qmd` for multiple publications. Each `qmd` file comes with meta data about the publication in the front matter. I can now easily list all publications, e.g. with

```yaml
listing: 
    contents: 
        - "publications/*.qmd"
```

I now want to provide a different view on this listing, i.e. restrict the listing to include only publications of a specific author. I can do this with a custom listing and an ejs template which filters on pattern `"NAME"`:

```ejs

<ul class="pub-list list">
  <% for (const item of items) { %>
    <% if(item.author.match("NAME") ){ %>
      <li <%= metadataAttrs(item) %>>
        <span class="listing-title"><%= item.title %>.</span>
      </li>
    <% } %>
  <% } %>
</ul>
```

I'm wondering if there is a way to 

1. retrieve the `"NAME"` pattern from the meta data of the page which is contains the listing, or
2. a different way to pass down variables to ejs templates?

**Comment by dragonstyle** [comment]
Right now there isn't a way to retrieve metadata from within the ejs template.

I think that would be a great feature (perhaps a utility function you could use to read metadata) but I would also like to add native filtering to listings. I'll open an issue to track these improvements.

**Reply by rcannood** [reply]
Thanks for taking a look at this! Is there already an issue for tracking these improvements?

**Reply by dragonstyle** [reply]
Yep https://github.com/quarto-dev/quarto-cli/issues/2833 was the issue tracking this- it should be supported as of Quarto 1.3.

**Comment by mcanouil** [comment]
To note, this is partially why I went another way to create a publication listing (*WIP*), _i.e._, I used pandoc to get a YAML list of publication from a bib file and use that yml list directly in the listings: <https://github.com/mcanouil/mickael.canouil.fr/blob/fix-issue23/publications.qmd>

![Screenshot of a custom listing for scientific publication list showing the number of main and other publication in the title of the page](https://user-images.githubusercontent.com/8896044/195626191-e5dde517-f03c-4162-92c9-bb09eabcb07c.jpeg)
